src/ed25519/hasher512.cpp

```diff
@@ -70,11 +78,40 @@ namespace libtorrent::aux {
 		gcry_md_copy(&m_context, h.m_context);
 		return *this;
 	}
+#elif defined TORRENT_USE_LIBCRYPTO
+	hasher512::hasher512(hasher512 const& h)
+		: hasher512()
+	{
+		EVP_MD_CTX_copy_ex(m_context, h.m_context);
```

Comment:
this is a bit of a nit-pick, but it seems `EVP_MD_CTX_copy()` is a better match here, since it's both initializes *and* copies the context. then you don't need to also invoke the default constructor.

Reply:
I had the same thought until I noticed it is viewed as obsoleted:

https://www.openssl.org/docs/manmaster/man3/EVP_MD_CTX_copy.html
>The functions EVP_DigestInit(), EVP_DigestFinal() and EVP_MD_CTX_copy() are obsolete but are retained to maintain compatibility with existing code. New applications should use EVP_DigestInit_ex(), EVP_DigestFinal_ex() and EVP_MD_CTX_copy_ex() because they can efficiently reuse a digest context instead of initializing and cleaning it up on each call and allow non default implementations of digests to be specified.